source/common/common/utility.cc

```diff
@@ -323,6 +323,33 @@ std::vector<absl::string_view> StringUtil::splitToken(absl::string_view source,
   return absl::StrSplit(source, absl::ByAnyChar(delimiters), absl::SkipEmpty());
 }
 
+std::string StringUtil::removeTokens(absl::string_view source, absl::string_view delimiters,
+                                     std::set<absl::string_view> tokens_to_remove,
+                                     absl::string_view joiner, bool trim_tokens, bool ignore_case) {
+  const auto values = Envoy::StringUtil::splitToken(source, delimiters);
+  std::string new_value;
+  for (auto& v : values) {
+    absl::string_view token_for_compare = v;
+    absl::string_view token_for_result = v;
+    if (trim_tokens) {
+      token_for_result = token_for_compare = StringUtil::trim(v);
+    }
+    std::string lower_value;
+    if (ignore_case) {
+      lower_value = StringUtil::toLower(token_for_compare);
+      token_for_compare = lower_value;
+    }
+    if (tokens_to_remove.count(token_for_compare) != 0) {
+      continue;
+    }
+    if (!new_value.empty()) {
+      new_value.append(joiner.data(), joiner.size());
+    }
+    new_value.append(token_for_result.data(), token_for_result.size());
```

Comment:
I believe keep a `std::vector<absl::string_view>` and `absl::StrJoin` at the end is more efficient.